src/main/cluster.ts

```diff
@@ -127,19 +127,20 @@ export class Cluster implements ClusterModel, ClusterState {
   }
 
   protected bindEvents() {
-    logger.info(`[CLUSTER]: bind events`, this.getMeta());
-    const refreshTimer = setInterval(() => !this.disconnected && this.refresh(), 30000); // every 30s
-    const refreshMetadataTimer = setInterval(() => !this.disconnected && this.refreshMetadata(), 900000); // every 15 minutes
-
-    if (ipcMain) {
-      this.eventDisposers.push(
-        reaction(() => this.getState(), () => this.pushState()),
-        () => {
-          clearInterval(refreshTimer);
-          clearInterval(refreshMetadataTimer);
-        },
-      );
+    if (!ipcMain) {
+      return
```

Comment:
Why this is needed? `cluster.activate() -> cluster.bindEvents()` called only at main AFAIK?

Reply:
We cannot be sure where this is called if/when `Cluster` is exposed to extensions.

Reply:
Well, maybe then it should be inside `cluster.activate()`?

Reply:
And `bindEvents()` actually is protected method which is not intended to call outside..

Reply:
I'll revert this change (not needed for this PR).. I think we have to refactor `Cluster` class quite a bit to make it safe/usable in both main & renderer contexts.